internal/net/net.go

```diff
@@ -29,12 +29,12 @@ func IsIPV6(ip _net.IP) bool {
 
 // IsPortAvailable checks if a TCP port is available or not
 func IsPortAvailable(p int) bool {
-	ln, err := _net.Listen("tcp", fmt.Sprintf(":%v", p))
+	conn, err := _net.Dial("tcp", fmt.Sprintf(":%v", p))
 	if err != nil {
-		return false
+		return true
 	}
-	ln.Close()
-	return true
+	defer conn.Close()
+	return false
```

Comment:
this function is being called in `cmd//nginx/flags.go` when Nginx has not started yet, which results in for the dial to fail. And consequently
```

F0803 14:42:19.694324       5 main.go:72] Port 80 is already in use. Please check the flag --http-port
```